lib/Dialect/SV/Transforms/HWLegalizeModules.cpp

```diff
@@ -154,26 +386,12 @@ void HWLegalizeModulesPass::processPostOrder(Block &body) {
     }
 
     if (options.disallowPackedArrays) {
-      // Try idioms for lowering array_get operations.
-      if (auto getOp = dyn_cast<hw::ArrayGetOp>(op))
-        if (auto *replacement = tryLoweringArrayGet(getOp)) {
-          it = Block::iterator(replacement);
-          anythingChanged = true;
-          continue;
-        }
```

Comment:
Probably we need to update iterator position like `it = Block::iterator(replacement);` to visit all operations generated within `tryLoweringPackedArrayOp`. Basically we have to set the iterator to a last operation created in`tryLoweringPackedArrayOp` because `processPostOrder` traverses in the inverse order.

Reply:
Yeah, this will be required for handling packed arrays with more dimensions. Currently the code only works if we _don't_ do this as we emit some assignments that we don't handle, but it's surely better to add support for those now. I'll look into it.

Reply:
Fixed this now. More array dimensions still don't work, but I may address that in a later PR.